util/suffix.c

```diff
@@ -69,6 +69,47 @@ const char *suffix_si_get(double *value)
 	return "";
 }
 
+static struct si_suffix_ld {
+	long double magnitude;
+	const char *suffix;
+} si_suffixes_ld[] = {
+	{1e30, "Q"},
+	{1e27, "R"},
+	{1e24, "Y"},
+	{1e18, "E"},
+	{1e15, "P"},
+	{1e12, "T"},
+	{1e9, "G"},
+	{1e6, "M"},
+	{1e3, "k"},
+	{1e0, ""},
+	{1e-3, "m"},
+	{1e-6, "u"},
+	{1e-9, "n"},
+	{1e-12, "p"},
+	{1e-15, "f"},
+	{1e-18, "a"},
+	{1e-21, "z"},
+	{1e-24, "y"},
+	{1e-27, "r"},
+	{1e-30, "q"},
+	{0}
+};
+
+const char *suffix_si_get_ld(long double *value)
```

Comment:
If we process 128 bit values, we should use either uint128_t or int128_t. `long double` is the wrong type IMO

Reply:
BTW,  this is also what I don't like on the other `suffix_*` functions.

Reply:
suffix_si_get converts decimal values as for example from 0.001 to 1m or from 0.000001 to 1u etc. also so double or long double values are necessary but not used uint128_t or int128_t integer types I think.

Reply:
My argument is that we have implicit type conversion which can be trouble. If we can avoid those, it's better solution in the long term.

Reply:
Regarding the 128bit types, isn't it not guaranteed to be available with many compiler+arch combinations? The long-double usage was initially a compromise to have an easily printable value with acceptable precision.

Reply:
We have defined our own uint128_t type. As we don't do any real arithmetic operation on this type we can get away with a few simple hacks. Basically we just need read and print operations. This even works on 32bit platforms.

Reply:
https://github.com/linux-nvme/nvme-cli/blob/c045535fb0e0e3d3059ed65f18ff9f541e2ad0ac/util/types.h#L20
https://github.com/linux-nvme/nvme-cli/blob/c045535fb0e0e3d3059ed65f18ff9f541e2ad0ac/util/types.c#L11